src/common/btree.c

```diff
@@ -3444,6 +3444,27 @@ dbtree_close(daos_handle_t toh)
 	return 0;
 }
 
+/**
+ * Reset daos open tree handle, mainly used during upgrade, since the container
+ * root might be changed due to the durable format change.
+ *
+ * \param[in]	toh	Tree open handle.
+ * \param[in]	root	the root to be set.
+ * \param[in]	root_off the root offset to be set.
+ */
+void
+dbtree_handle_reset_root(daos_handle_t toh, struct btr_root *root, umem_off_t root_off)
+{
+	struct btr_context	*tcx;
+
+	tcx = btr_hdl2tcx(toh);
+	if (tcx == NULL)
+		return;
+
+	tcx->tc_tins.ti_root = root;
+	tcx->tc_tins.ti_root_off = root_off;
```

Comment:
Minor: you could handle this internally rather than passing extra argument, no?

Reply:
these tree structure (ti_root/ti_root_off) are internally to btree. not sure how to do this without passing the arguments.

Reply:
ti_root and ti_root_off are the same thing.   root is the virtual address and root_off is the offset. You should be able to do something like

umem_ptr2offset(....) on root to get what root_off should be.